src/common/etcd_delegate.rs

```diff
@@ -111,12 +116,96 @@ impl EtcdDelegate {
         })?;
 
         let kvs = resp.take_kvs();
+
         match kvs.get(0) {
-            Some(kv) => Ok(Some(util::decode_from_bytes::<T>(kv.value())?)),
+            Some(kv) => Ok(Some((
+                util::decode_from_bytes::<T>(kv.value())?,
+                kv.version(),
+            ))),
             None => Ok(None),
         }
     }
 
+    /// offer an version check
+    async fn write_to_etcd_with_version<
+        T: DeserializeOwned + Serialize + Clone + Debug + Send + Sync,
```

Comment:
Just a suggestion: Perhaps we should replace these template constraints with &[u8] instead. This would result in faster compilation speed since templates wouldn't be required.

Reply:
I think that's not necessary.

Reply:
Currently it seems not necessary, but we use generic to make it extendable. So I prefer to keep it that way.